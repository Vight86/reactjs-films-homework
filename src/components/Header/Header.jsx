import React from 'react';
import propTypes from 'prop-types';

import styles from './Header.scss';

const Header = ({ className, children }) => (
  <header className={styles[className]}>
    {children}
  </header>
);

Header.propTypes = {
  className: propTypes.string,
  children: propTypes.node,
};

Header.defaultProps = {
  className: '',
  children: <span />,
};

export default Header;
