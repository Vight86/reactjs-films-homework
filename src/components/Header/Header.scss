@import '../../utils/variables';

.header {
  display: flex;
  box-sizing: border-box;
  justify-content: space-between;
  padding-right: $pr;
  padding-left: $pl;
}

.primary {
  @extend .header;

  height: 15em;
  padding-top: 3em;
  background: linear-gradient(to bottom, #8a8b8b, transparent);
}

.secondary {
  @extend .header;

  padding: 0;
  border-bottom: 1px solid transparentize($color: $text-tertiary, $amount: 0.9);
  margin-bottom: 4em;
}
