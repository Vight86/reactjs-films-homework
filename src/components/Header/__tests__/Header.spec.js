import React from 'react';
import ShallowRenderer from 'react-test-renderer/shallow';
import Header from '../index';

let renderer;
let result;

beforeEach(() => {
  renderer = new ShallowRenderer();
  renderer.render(
    <Header className="test">
      <h1>title</h1>
    </Header>,
  );
  result = renderer.getRenderOutput();
});

afterEach(() => {
  renderer = null;
  result = null;
});

describe('render Header component', () => {
  it('render correctly', () => {
    expect(result).toMatchSnapshot();
  });
});
