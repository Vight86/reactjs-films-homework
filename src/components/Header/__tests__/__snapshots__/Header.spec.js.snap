// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render Header component render correctly 1`] = `
<header
  className="test"
>
  <h1>
    title
  </h1>
</header>
`;
